### DISCUSSION TITLE
Terminating Boa runtime after timeout

### DISCUSSION THREAD

**Discussion by mlafeldt** [boa-dev/boa#3238] [Q&A]
I want to implement a timeout limit similar to https://github.com/mlafeldt/deno-limits/tree/main/src

I know that Boa has a loop iteration limit. However, being able to stop the runtime after X seconds would be much more useful since it's hard to configure a sensible iteration limit in advance.

With Deno, I can grab the V8 isolate's handle and terminate the isolate after a timeout (see link above). I wonder if there's any way to do the same with Boa? If not, is it safe to terminate a thread that executes JS using Boa?

**Comment by jedel1043** [comment]
Not currently. Also, I don't know if it would be safe to terminate the thread containing a Boa instance, since I think GC won't deallocate the used space if the thread suddenly stops without running the object's destructors.
